weave

```diff
@@ -178,12 +178,19 @@ if [ "$1" = "run" ] ; then
     exec_remote attach $CIDR_ARGS --or-die $CONTAINER >/dev/null
     echo $CONTAINER
     exit 0
-elif [ "$1" != "--local" ] ; then
+elif [ "$1" = "env" -a "$2" = "--restore" ] ; then
+    true
+elif [ "$1" = "--local" ] ; then
+    shift
+else
     exec_remote "$@"
     exit $?
 fi
 
-shift 1
+if [ "$1" = "env" -a "$2" = "--restore" ] ; then
```

Comment:
could just put this ahead of the `if [ "$1" = "run" ]` and remove the extra shenanigans.